sql/src/main/java/io/crate/planner/PlanNodeBuilder.java

```diff
@@ -120,6 +141,10 @@ public static CollectNode collect(TableInfo tableInfo,
         node.projections(projections);
         node.isPartitioned(tableInfo.isPartitioned());
         setOutputTypes(node);
+        node.orderBy(orderBy);
+        if (limit != null) {
+            node.limit(limit);
```

Comment:
setter could be made Nullable as limit() on the CollectNode can be Null anyway. So you can avoid the null check